src/picongpu/include/particles/traits/GetMarginPusher.hpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2015 Richard Pausch
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+#include "traits/GetMargin.hpp"
+#include "particles/traits/GetInterpolation.hpp"
+#include "particles/traits/GetPusher.hpp"
+
+
+namespace picongpu
+{
+
+namespace traits
+{
+template<typename T_Species>
+struct LowerMarginPusher
+{
+  typedef typename PMacc::math::CT::add<  
+    typename GetLowerMargin< typename GetInterpolation< T_Species >::type >::type,
+    typename GetLowerMargin< typename GetPusher< T_Species >::type >::type
+    >::type type;
+};
```

Comment:
Better write this in terms of apply:

    typedef PMacc::math::CT::add< GetLowerMargin< GetInterpolation< _1 > >, GetLowerMargin< GetPusher< T_Species > > > AddMargins;
    typedef typename bmpl::apply<AddMargings, T_Species>::type;

Also don't use GetLowerMargin for the pusher but GetLowerMarginExtension as otherwhise its wrong or misleading